Example downloads & reader/writer updates

banesullivan opened the pull request:
This adds the ability to download a few examples from the VTK examples repository.

It also cleans up some to the experimental universal readers code.

### Todo

- [ ] all readers need to be tested
- [x] example downloads need to be tested?
- [x] should examples download to temp folder or to working directory?
- [ ] a second dictionary of `attrs` and arguments needs to be created for readers that do not work out of the box
- [x] Fix issue with windows `urllib` bug

### Example

```py
import vtki
from vtki import examples

head = examples.download_head()

tool = vtki.OrthogonalSlicer(head)

vol = head.contour([15])
tool.plotter.add_mesh(vol, color='yellow', name='skin', opacity=0.5)

```

![foo](https://user-images.githubusercontent.com/22067021/53694019-c2297980-3d65-11e9-9c18-7f401f56c1c0.png)


### Note

Including tests for the downloads might not be a good idea... it slows down the testing suite significantly.

akaszynski commented:
I think it’s probably best to include testing for the downloadable examples.  It’s how I feel about appveyor.  It’s easy to just exclude it, but it can easily be forgotten with API changes and you’ll end up with outdated examples.

banesullivan commented:
Fair point - I just added a conditional so that the downloads tests will run if there is a `TEST_DOWNLOADS` environmental variable set on the operating system. 

I went ahead and added that variable to Travis and Appveyor so that the tests get executed but we don't get bogged down by the downloads when testing locally.

akaszynski commented:
Good idea.  I run tests before I commit using a pew-commit hook; stops me from adding junk code.  I have a separate set of tests, called functional_tests, for releases on my closed source code.  However, for online CI, it’s no cost; I usually just grab a coffee.
